Updating eval-rst blocks to myst directives

choldgraf opened the pull request:
There are `{eval-rst}` blocks scattered throughout the documentation that should now be convertible to pure MyST markdown directives. This is a PR to take a first pass at this.

choldgraf committed:
Updating jsoninclude directives to myst

choldgraf committed:
Converting toctrees to myst directives

choldgraf committed:
CSV table no translate directive to myst

choldgraf committed:
Converting two more eval_rst directives

choldgraf committed:
Adding extension for toctree item that was missed.

choldgraf committed:
Converting note admonitions to MyST

choldgraf commented:
OK I think this one is ready for review. There are more `eval-rst` blocks in there, but this knocks out a bunch of them, and touches enough parts of the documentation that it's probably worth merging sooner than later once it looks correct (I already had to do a rebase that required a few conflict resolutions). @jpmckinney I'm not sure what is the right "test" to use to determine if it's mergeable...the tests are passing but I feel that ultimately it's the "eye test" that is most important. Maybe there are certain sections you'd be particularly concerned about being messed up in the move to MyST or something?

jpmckinney commented on docs/guidance/map.md:
```diff
@@ -120,11 +118,10 @@ Some data elements might not match any field or code in OCDS. To cover such case
+:hidden:
 
-   map/extensions
+map/extensions.md
```
Why does this line need `.md` but other toctrees don't?

choldgraf replied:
To be honest I'm not sure, in this case it was throwing a warning and the only reason I could come up with was that there was both a file called map and a folder called map

jpmckinney commented on docs/guidance/map/awards_contracts_buyers_suppliers.md:
````diff
@@ -16,12 +16,7 @@ The [UNCITRAL Model Law on Public Procurement (2011)](https://uncitral.un.org/en
-
-   .. markdown::
-
+```{note}
````
Is this the same output as:

```
.. admonition:: Note
   :class: note
```

choldgraf replied:
That's my understanding, but I can double check if need be. Just seemed like a simplification without changing the output but if does then i can revert

jpmckinney commented on docs/guidance/map/awards_contracts_buyers_suppliers.md:
````diff
@@ -36,55 +31,36 @@ Contracting processes can result in different types of contract between buyers a
-    .. markdown::
-
-      Electronic catalogs can often be found in more complicated procedures, usually together with framework agreements and involving multiple bidders. However, on their own, they are simply an electronic format (typically prescribed by the buyer) that participants in the contracting process must follow when exchaning information about technical specifications, evaluation criteria, bids, lots, et ...
+```{note}
````
The original was a hint admonition. On the other hand, I think this content is fine as a note.

jpmckinney commented on docs/schema/codelists.md:
````diff
@@ -6,13 +6,8 @@ Codelists can either be open or closed. **Closed codelists** are intended to be
-   .. markdown::
-
-      If you use new codes outside those in an open codelist, please document the codes in an [OCDS extension](../guidance/map/extensions) and in your [publication policy](../../guidance/publish/#finalize-your-publication-policy). Please also create an issue in the [OCDS GitHub repository](https://github.com/open-contracting/standard/issues), so that the codes can be considered for inclusion in t ...
+```{admonition} Extending open codelists
````
Will this admonition display as before, with class note?

jpmckinney replied:
Hmm, no, it doesn't. https://standard.open-contracting.org/staging/eval-rst/en/schema/codelists/

jpmckinney commented on docs/guidance/map/organizational_units.md:
````diff
@@ -40,30 +37,23 @@ In Moldova, the national procurement agency needs to include a division code for
+```{jsoninclude} ../../examples/organizational-units/ocds_divisionCode_extension/extension.json
+:jsonpointer: 
+:expand: name, description, schemas, compatibility
+   :title: exension
````
Typo

jpmckinney commented on docs/guidance/map/organizational_units.md:
````diff
@@ -40,30 +37,23 @@ In Moldova, the national procurement agency needs to include a division code for
+```{jsoninclude} ../../examples/organizational-units/ocds_divisionCode_extension/release-schema.json
+:jsonpointer: 
+:expand: definitions, Organization, properties, details, properties, divisionCode
+   :title: reease-schema
````
Typo

jpmckinney commented on docs/guidance/map/pre-qualification.md:
````diff
@@ -166,10 +153,9 @@ The invitation represents the initiation of a contracting process to establish a
-   :title: Tender section
+```{jsoninclude} ../../examples/pre-qualification/pre-qualification-paraguay-package.json
+:jsonpointer: /releases/0/tender
+   :title: Tnder section
````
Typo

jpmckinney reviewed (commented):
I assume you did some sort of regex-replace? I checked for form, rather than for specific directive arguments being identical. I did notice a few typos introduced, which makes me a bit wary (see comments). Can you explain how they might've occurred?

I also just noticed that the `{download}` role causes the word Schema on this page to be bold: https://standard.open-contracting.org/staging/eval-rst/en/schema/release/ However, once we change the theme the problem will probably be fixed.

jpmckinney commented on docs/guidance/map/awards_contracts_buyers_suppliers.md:
````diff
@@ -16,12 +16,7 @@ The [UNCITRAL Model Law on Public Procurement (2011)](https://uncitral.un.org/en
-   .. markdown::
-
+```{note}
       The OCDS schema and documentation are not clear what, precisely, is meant by 'award'. A revision of the definition of `Award` in OCDS is being considered for a future version of the standard ([GitHub issue](https://github.com/open-contracting/standard/issues/895)).
````
The indentation of all admonition/note/warning blocks needs to be fixed. Otherwise, they appear as code blocks, like https://standard.open-contracting.org/staging/eval-rst/en/guidance/map/awards_contracts_buyers_suppliers/

choldgraf commented:
Thanks for this review, I will take another pass and clean up the missing pieces. I did a semi automated conversion, so it wasn't just a regex replace but it happened fast enough that I could have missed some things because it was a lot of files.

jpmckinney committed:
myst: Use csv-table instead of codelisttable, open-contracting/sphinxcontrib-opencontracting#22

choldgraf committed:
Addressing review comments and another typo pass

choldgraf committed:
Converting a few more admonitions

choldgraf commented:
OK I addressed each of your comments above, and did another pass to try and find any other errors. Also converted a couple straggling admonitions. I don't think there are any typos there but you can see the changes in the latest commits if you'd like to check.

jpmckinney commented:
Replaced by #1247.
